webapp/src/ts/modules/contacts/contacts-content.component.ts

```diff
@@ -237,15 +238,35 @@ export class ContactsContentComponent implements OnInit, OnDestroy {
     const reportStartDate = months ? moment().subtract(months, 'months') : null;
 
     const allReports = reports || this.selectedContact?.reports || [];
-    this.filteredReports = allReports
-      .filter((report) => !reportStartDate || reportStartDate.isBefore(report.reported_date));
+
+    this.filteredReports = this.getFilteredReports(allReports, reportStartDate, this.DISPLAY_LIMIT);
+  }
+
+  private getFilteredReports(allReports: any[], reportStartDate, displayLimit): any[] {
+    const filteredReports: any[] = [];
+    for (const report of allReports) {
+      if (filteredReports.length >= displayLimit) {
+        break;
+      }
+
+      if (this.validateReport(report, reportStartDate)) {
```

Comment:
```suggestion
      if (reportStartDate?.isBefore(report.reported_date)) {
```
Can you do this and then delete `validateReport `?